File: CurtinhasBackEnd/CurtinhasBackEnd2/Models/Entidades/Curtinha.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace CurtinhasBackEnd.Models.Entidades
{
    public class Curtinha
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Url da imagem é obrigatório")]
        public string UrlImagem { get; set; }

        [Required(ErrorMessage = "Título é obrigatório")]
        [StringLength(200, MinimumLength = 3, ErrorMessage = "Título deve ter entre 3 e 200 caracteres!")]
        public string Titulo { get; set; }

        [Required(ErrorMessage = "Resumo é obrigatório")]
        [StringLength(500, MinimumLength = 3, ErrorMessage = "Resumo deve ter entre 3 e 500 caracteres!")]
        public string Resumo { get; set; }

        public string Detalhes { get; set; }

        public DateTime DataPublicacao { get; set; }

        public DateTime DataEdicao { get; set; }

        [Required(ErrorMessage = "Link é obrigatório")]
        public string Link { get; set; }
    }
}

File: CurtinhasFrontEnd/src/app/models/curtinha.ts
export class Curtinha {
  id: number;
  urlImagem: string;
  titulo: string;
  resumo: string;
  detalhes: string;
  dataPublicacao: Date;
  dataEdicao: Date;
  link: string;

  constructor(
              id: number,
              urlImagem: string,
              titulo: string,
              resumo: string,
              detalhes: string,
              dataPublicacao: Date,
              dataEdicao: Date,
              link: string) {

      this.id = id;
      this.urlImagem = urlImagem;
      this.titulo = titulo;
      this.resumo = resumo;
      this.detalhes = detalhes;
      this.dataPublicacao = dataPublicacao;
      this.dataEdicao = dataEdicao;
      this.link = link;
  }
}
